### ISSUE TITLE
no-unmodified-loop-condition reports even loop is interruptible (by yield)

### ISSUE THREAD

**Issue by polomsky** [eslint/eslint#8496]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 6.9.1
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

```js
module.exports = {
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
    "rules": {
        "no-unmodified-loop-condition": "error",
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
var aborted = false;

function abort() {
    aborted = true;
}

function* looper() {
    while (!aborted) {
        yield someMethod();
    }
}

module.exports = {
    abort,
    looper,
}

```

**What did you expect to happen?**
I expected that error is not reported. I am not sure if this is actually bug but for me is false positive.
Loop variable is visible outside loop scope (and actually can be modified) and loop itself is in generator method.

**What actually happened? Please include the actual, raw output from ESLint.**
`error  'aborted' is not modified in this loop  no-unmodified-loop-condition`

**Comment by gyandeeps** [comment]
Considering the static nature of eslint, we only operate within the module boundary. Based on your example, its not getting modified inside this particular module.
I would recommend disabling this rule for that particular line.

**Comment by polomsky** [comment]
I can not image better alternative.
